src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -1255,6 +1285,19 @@ protected void wipeOfTestResources(
         }
     }
 
+    protected void validateDocCountAndEmbedding(String indexName, int expectedDocCount, String docId, final String embeddingField) {
+        int count = getDocCount(indexName);
+        assertEquals(expectedDocCount, count);
+        Map<String, Object> document = getDocById(indexName, docId);
```

Comment:
How about using the `getFirstInnerHit` function? Besides, do we need to validate every document from the index?

Reply:
Single doc retrieval doesn't have the `hits` field in response. We verify the total count first that means we successfully ingest all the documents, then we pick up one doc to verify it's detailed information, doesn't seem necessary to verify them all.

Reply:
Agree that we can only verify a single document. For `getFirstInnerHit`, we can query the index as the following. This saves us querying the document with the document id.
```
Map<String, Object> searchResults = search(indexName, query, 10);
assertNotNull(searchResults);
Map<String, Object> document = getFirstInnerHit(searchResults);
```

Reply:
I get your point, that should also work. The reason why I feel I need the document id is that I create documents with ids in order e.g. "0", "1", "2", "3", "4" etc. I verify the last document through the last id so that I can verify not a random one but the specific one and it should be there and not be cut off.

Reply:
Is it a hard requirement to verify a specific document? From my understanding, verifying any document from the index makes sense.

Reply:
For example, some other test unexpectedly reuses your index and stores 10 documents but your tests fail to store documents because of some bugs, but your verification can pass because you still get 10 documents through query. By querying doc of specific id, although similar problem mentioned above is not eliminated, but you reduce the chance to pretty low.

Reply:
Making sense. I will resolve the conversation.